qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/CreateIdentityUITest.java

```diff
@@ -68,28 +75,50 @@ public void setUp() throws Exception {
         wakeLock = UIActionHelper.wakeupDevice(mActivity);
         mSystemAnimations = new SystemAnimations(mActivity);
         mSystemAnimations.disableAll();
+
+        //TODO TMP Fix option
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            InstrumentationRegistry.getInstrumentation().getUiAutomation().executeShellCommand(
+                    "pm grant " + InstrumentationRegistry.getTargetContext().getPackageName()
+                            + " android.permission.READ_PHONE_STATE");
+        }
     }
 
 
     @Test
     public void testCreateIdentity() throws Throwable {
-        String identity = "spoon2";
-        createIdentityPerformEnterName(identity);
+        createIdentityPerformEnterName();
+        createIdentityEnterEmail();
+        createIdentityEnterPhone();
         createIdentityPerformConfirm();
     }
 
-    private void createIdentityPerformEnterName(String identity) throws Throwable {
+    private void createIdentityPerformEnterName() throws Throwable {
         onView(withText(R.string.create_identity_create)).check(matches(isDisplayed())).perform(click());
-        onView(allOf(withClassName(endsWith("EditTextFont")))).perform(setText(identity), pressImeActionButton());
         onView(withText(R.string.create_identity_enter_name)).check(matches(isDisplayed()));
-        closeSoftKeyboard();
-
         UITestHelper.screenShot(UITestHelper.getCurrentActivity(mActivity), "input");
+        onView(withId(R.id.et_name)).perform(setText(name));
         onView(withText(R.string.next)).perform(click());
     }
 
+    private void createIdentityEnterEmail() throws Throwable {
+        onView(withText(R.string.create_identity_enter_email)).check(matches(isDisplayed()));
```

Comment:
This gets repeated in the android project a couple of times. If it's an assertion, than you can maybe extract some general utility functions here.
`assertView withText(R.string.create_identity_enter_mail)`
or an extracted `assertDisplayed(R.string.create_identity_final)`
or whatever fit's to the android project should be a neat refactoring.

Page objects could also clean those tests up to separate real assertions from those made to debug failure. The `testCreateIdentity` could look like
```JAVA
public void testCreateIdentity() throws Throwable {
    CreateIdentityView view = ...;  // the page object
    view.enterName("new name")
        .enterEmail("test@example.com")
        .enterPhone("12345")
        .confirm();

    Identity identity = ...; // get newly created identity
    identity.name shouldMatch "new name"
    identity.email shouldMatch "test@example.com"
    identity.phone shouldMatch "12345"
}
```
And the "enterEmail" method could cover all those "onView...check(matches(isDisplayed()))" while the test clearly shows what's done. I use the dialog to add an identity (act) and then I check that the new identity contains what I told it to (assert). (Not sure if that's what your test does because I can't tell by looking at it).